infra/testing/comlink/sw-interface.js

```diff
@@ -0,0 +1,50 @@
+importScripts('/comlink.js');
+
+// TODO: Standardize on naming, and move over some of the legacy uses of
+// postMessage() to this new approach.
+// These are all the methods that will be called in the SW, but are exposed in
+// the window context via Comlink.
+const api = {
+  cachesKeys: () => {
+    return caches.keys();
+  },
+
+  clearAllCaches: async () => {
+    const keys = await caches.keys();
+    return Promise.all(keys.map((key) => caches.delete(key)));
+  },
+
+  doesDbExist: (dbName) => {
+    return new Promise((resolve) => {
+      const result = indexedDB.open(dbName);
+      result.onupgradeneeded = (event) => {
+        event.target.transaction.abort();
+        event.target.result.close();
+        resolve(false);
+      };
+      result.onsuccess = (event) => {
+        event.target.result.close();
+        resolve(true);
+      };
+    });
+  },
+
+  cacheContents: async (cacheName) => {
```

Comment:
Would it make more sense to call this `cacheUrls` or `cacheRequestUrls` since it returns the request URL rather than the request body?

Reply:
Done.